### PR TITLE
Add adp analytics to Dynamo

### PR DESCRIPTION
**Pull request by pinzart**
### Purpose

Add ADP analytics to Dynamo.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs [hunk 1]
```diff
@@ -36,6 +38,9 @@ public void Start(DynamoModel model)
             if(service.GetTrackerFactory(GATrackerFactory.Name) == null)
                 service.Register(new GATrackerFactory(ANALYTICS_PROPERTY));
 
+            if (service.GetTrackerFactory(ADPTrackerFactory.Name) == null)
+                service.Register(new ADPTrackerFactory());
```
Comment (mjkkirschner) [inline src/DynamoCore/Logging/DynamoAnalyticsClient.cs:42]: can this fail for some reason?

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs [hunk 2]
```diff
@@ -169,7 +175,7 @@ public DynamoAnalyticsClient(DynamoModel dynamoModel)
             //Dynamo app version.
             var appversion = dynamoModel.AppVersion;
 
-            product = new ProductInfo() { Name = "Dynamo", VersionString = appversion, AppVersion = dynamoModel.Version };
+            product = new ProductInfo() { Id = "DYN", Name = "Dynamo", VersionString = appversion, AppVersion = dynamoModel.Version, BuildId = "CB159AA3-2F20-4791-AE8C-8939E90C9600", ReleaseId = "2.5.0", MasterId = "{4C7B248C-56D3-488F-8E23-8CAFD2651968}" };
```
Comment (mjkkirschner) [inline src/DynamoCore/Logging/DynamoAnalyticsClient.cs:178]: is this going to effect the gathered google analytics data?

Comment (mjkkirschner) [inline src/DynamoCore/Logging/DynamoAnalyticsClient.cs:178]: whats the process around these ids and when we increment them?
Latest released version is 2.5.3

**Comment by mjkkirschner** [comment]
Should we also update our readme and associated terms of use etc before this gets merged into master?
@kronz 
@Amoursol

**Comment by QilongTang** [comment]
@mjkkirschner The about box already covers ADP. We are talking with legal about this, this is not meant for merging.

**Comment by pinzart** [comment]
Sorry @mjkkirschner ..I should have made it clear in the description.
We should test this branch internally, with the ADPC#SDK dll. Check that we have the proper analytics showing up in Looker...then we can merge to master.

We've just talked to legal. It seems it will take a while to get proper approvals (including updating the terms of use ...if needed)

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs (hunk 1 above)
Comment (pinzart) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:42]: not really. At the moment...there is no more risk than the GATracker registration call.
We can wrap bot calls in a try catch...

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs (hunk 2 above)
Comment (pinzart) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:178]: The added data should not interfere with Google analytics.

It is still designed to simplify internal testing (just test dummy identifiers). 
We need to decide on proper UPI (universal product identifier) configs ...then we can properly supply them here

File: src/Dynamo.All.sln [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.28010.2050
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.29806.167
```
Comment (QilongTang) [inline src/Dynamo.All.sln:4]: If possible, would you revert this change as well as the version change above? These are local IDE and timestamp dependent

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs (hunk 2 above)
Comment (pinzart) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:178]: Again...this is test dummy data.
From my discussion with other teams....it seems that the proper releaseId format is the targeted release year (ex. 2021) So we would need to change the UPI configs
THe rest of the identifiers (from my understanding) do not play any real roles (they should just reflect the configs in UPI..and they do not need to change from release to release)

Basically it sounds like the ReleaseId is the only config that needs to change from release to release.

File: src/Dynamo.All.sln (hunk 3 above)
Comment (pinzart) [reply src/Dynamo.All.sln:4]: yes

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs (hunk 2 above)
Comment (QilongTang) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:178]: @mjkkirschner We update them everytime we RTM. They need to be first populated from UPI and then reflected in Code here to be passed in. These will need to be documented when this gets in.

**Comment by Amoursol** [comment]
100% yes, we do need to update TOS and our legal messaging with this @mjkkirschner, thanks for the ping. 

@QilongTang @pinzart Am I correct in reading that you two are talking with legal (James Connor?) on this?

**Comment by QilongTang** [comment]
@Amoursol Yes, we already in connect with legal. For TOU, do you mean aboutbox only or if there is something else we need to watch for? I am not aware of anywhere else we mention Analytics component. In terms of agreement, we will not be using the DynamoGA one for ADP agreement.

**Comment by Amoursol** [comment]
@QilongTang - Brilliant, thank you. Yes, meant the About-Box in Dynamo's help menu 😊

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 4]
```diff
@@ -239,8 +239,8 @@
   <data name="CompactLayoutTitle" xml:space="preserve">
     <value>Compact</value>
   </data>
-  <data name="ConsentFormGoogleAnalyticsCheckBoxContent" xml:space="preserve">
-    <value>Yes, I would like to contribute to this program (Google Analytics).</value>
+  <data name="ConsentFormAdskAnalyticsCheckBoxContent" xml:space="preserve">
+    <value>Yes, I would like to contribute to this program (Autodesk Analytics).</value>
```
Comment (pinzart) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx:243]: not sure if it's ok to change existing resources ....

**Comment by pinzart** [comment]
Changed the analytics consent related documents/resource.
I intend to enable the ADP supplied consent window through a debug mode later on (if it makes sense)

**Comment by mjkkirschner** [comment]
@pinzart - the build is failing with:
``` 
 Configuration\DebugModes.cs(50,150): error CS1002: ; expected [c:\WorkspaceDynamo\src\DynamoCore\DynamoCore.csproj]
[2020-05-05T17:19:32.922Z] 
[2020-05-05T17:19:32.922Z]     226 Warning(s)
[2020-05-05T17:19:32.922Z]     1 Error(s)
[2020-05-05T17:19:32.922Z] 
```

**Comment by pinzart** [comment]
@mjkkirschner My fault...forgot a comma.
Should be fixed

File: src/DynamoCoreWpf/Properties/Resources.Designer.cs [hunk 5]
```diff
@@ -19,7 +19,7 @@ namespace Dynamo.Wpf.Properties {
     // class via a tool like ResGen or Visual Studio.
     // To add or remove a member, edit your .ResX file then rerun ResGen
     // with the /str option, or rebuild your VS project.
-    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "16.0.0.0")]
+    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "15.0.0.0")]
```
Comment (pinzart) [inline src/DynamoCoreWpf/Properties/Resources.Designer.cs:22]: no idea what caused this change...

Comment (QilongTang) [reply src/DynamoCoreWpf/Properties/Resources.Designer.cs:22]: hmm Can you revert this then?

Comment (pinzart) [reply src/DynamoCoreWpf/Properties/Resources.Designer.cs:22]: It's an auto generated file.
I can change it manually..

**Comment by pinzart** [comment]
@mjkkirschner @QilongTang are we good to go here? 
I will add a separate PR for the UI changes (opt in/out of Google and ADP analytics)

**Comment by mjkkirschner** [comment]
@pinzart @QilongTang @Dewb  - have we done any testing in hosts that also use ADP? Could there be conflicts? What can we do in those cases?

**Comment by QilongTang** [comment]
@mjkkirschner We confirmed that the wrapper will be in charge of finding the correct version of ADP Client. We do not need to worry about versioning for now
